### PR TITLE
Reworked zombie evolution and spawns

### LINKED ISSUE #23063: Wander spawn monsters don't follow evolution rules
Game version:0.C-26482-g5d60a3e

Operating system:Windows 7

Tiles or curses:curses

Mods active:Filthy clothing, disable NPC needs, simplified nutrition, generic guns, more survival tools, makeshift items, mining, cataclysm++, stats through skills IV, extra professions, classes and scenarios

### Expected behavior
Wander spawns follow evolution rules so that overpowering monsters don't appear until your evolution rate factor says so

### Actual behavior
Special zombies, especially the bionic zombie supersoldiers, start spawning right away--even when the evolution rate is set slower than normal.

### Steps to reproduce the behavior
Start a game, wander around, wait to be ganked by a Zombie Super BAMRU.

### PR DESCRIPTION
**Pull request by Firestorm01X2**
Redoing this: https://github.com/CleverRaven/Cataclysm-DDA/pull/27950
to mainline.

Built version of this PR:
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/20_04_2019

Reddit post:
https://www.reddit.com/r/cataclysmdda/comments/ba6xic/zombie_evolution_rework_needs_testers/

This PR aims to remove very advanced zombie monsters from the beginings of the game and let basic zombies evolve to advanced using evolution rules.

Long time ago evolution was impleneted. But monstergrups still contains advanced monsters in the beginning. 

That may produce advanced mosnters at the start of the game. It is espesially noticable with wander spawn enabled. 

There is issue based on it:
https://github.com/CleverRaven/Cataclysm-DDA/issues/23063

#### Summary

SUMMARY: Content "Changing monstergroup to remove very advanced zombie monsters from the beginings ot the game"

#### Purpose of change

Fixes #23063

https://github.com/CleverRaven/Cataclysm-DDA/issues/23063#issuecomment-392184567
>Horde monsters (aka wandering zombies) are spawned via the monster group "GROUP_ZOMBIE", which contains shocker, brute, master, hulk etc. (see "data/json/monstergroups.json").

>And the monsters are upgraded normally when spawned.

>Maybe the monster group should be changed to not contain advanced monsters? Upgrading the monsters upon spawning them is done for probably all spawning monsters, so changing that would have a larger effect.

https://www.reddit.com/r/cataclysmdda/comments/aefhos/did_the_city_layout_rework_change_the_spawn_or/
>kevingranade
>https://github.com/CleverRaven/Cataclysm-DDA/issues/23063 tl;dr we never actually completed the evolution migration where only vanilla zombies spawn at game start, so in particular with wandering zombies enabled you can randomly get more advanced zombies.

#### Describe the solution

Let follow some basic rules. In the beginning there is should be only non heavily mutated zombies.
Zombie count as "non heavily mutated" as long as it is not contain feature that is not common to human host. 

So normal zombes, slow zombies, fast zombies, tought zombes, profession zobmies, zombies wih malfunctoned CBM, zombes with grenades and flametwhowers, zombes with power armor are allowed.

But as long as zombie start develop things, like- bone exoskeleton, acid spewing, supernatural abilites then it is moved from beginning of the game. Evolution will bring it as time come.

So no acid zombies, hulks and brutes, zombes with extra long arms, exta sharp claws, invisiblility in the darkness etc,. Player will see them later because of evolution process.

But there are exeptions:
* Lab spawns are untouched. It is expected to have advanced zombies there from the start due to "expriements".
* There are special groups to spawn stong enemies. This groups are left untouched.

Important note: Obviously, it will make beginning of the game easier. 

But it is more then just removing advanced monsters from spawn list.

Overall idea:
* "I am suriving for one month and I am seeing that zombies are changing. I am starting to see very werid  creatures in the horde from time to time."
* Make evolution a lot more unpredictable but at the same time more smooth in longer perion of time.
* Advanced monsters no longer going to just appear, but have to evolve from basic forms. Probably there is going to be exceptions.
* Start is easier- mostly only mundane zombies on the streets. But it is not going to be that way forever - you will see new mosnters at the end of the firts month. Not a much of them, however.
* First mounth you have to deal with only mundane zombies **mostly**. Skeletal Juggernout actually can happen in first month but it should be **very** unlikely. But still - beware.

Current idea overview:
* All advanced zombies removed from starting street city groups and house groups.
* Evolution starts almost instantly without 2 months of waiting to even start seeing changes.
* Default average evolution time is near 2 months. As before. Base zombie evolution time lowerd by 2 times.
* Enabled back true half life evolution time, that was changed here: https://github.com/CleverRaven/Cataclysm-DDA/pull/14077 . So now time to evolve will be more random, without unnecessary 100% pass interval.
* Base street monster group ("GROUP_ZOMBIE") no longer will be transformed to another group "GROUP_ZOMBIE_MID" after some time. Zombies now acutally have to evolve.
* Removed one way shortcut to hulks and skeletal juggernouts. To get here zombes must evolve at least 2 times.
* Slightly more zombes on streets at start - more meat to evolve.
* Shocker zombie now have chance to evolve to shocker brute.
* Zombie dog now evolves to skeleton zombie dog.

#### Describe alternatives you've considered

Change only GROUP_ZOMBIE (default monster group) and leave others as it was before.

#### Additional context

None.

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Now it more than removing advanced monsters from spawn list.

Overall idea:
* "I am suriving for one month and I am seeing that zombies are changing. I am starting to see very werid  creatures in the horde from time to time."
* Make evolution a lot more unpredictable but at the same time more smooth in longer perion of time.
* Advanced monsters no longer going to just appear, but have to evolve from basic forms. Probably there is going to be exceptions.
* Start is easier- mostly only mundane zombies on the streets. But it is not going to be that way forever - you will see new mosnters at the end of the firts month. Not a much of them, however.
* First mounth you have to deal with only mundane zombies **mostly**. Skeletal Juggernout actually can happen in first month but it should be **very** unlikely. But still - beware.

Current idea overview:
* All advanced zombies removed from starting street city groups and house groups.
* Evolution starts almost instantly without 2 months of waiting to even start seeing changes.
* Default average evolution time is 2 months.
* Added ability to enable back true half life evolution time, that was changed here: https://github.com/CleverRaven/Cataclysm-DDA/pull/14077 . So now time to evolve will be more random, without unnecessary 100% pass interval.
* Instead added JSON value for guranteed time to evolve to prevent too rapid evolution of one mosnter.
* Base street monster group ("GROUP_ZOMBIE") no longer will be trnasformed to another group "GROUP_ZOMBIE_MID" after some time. Zombies now acutally have to  evolve.
* Removed one way shorcut to hulk and skeletal juggernouts. To get here zombes must evolve at least 2 times.
* Slightly more zombes on streets at start - more meat to evolve.
* Time to monster upgrade going to be set around 2 months. Current are way to scattered- between 10 days and month. It is hard to balance due random evolution roll. Also it will give explainable scaling with WORLD OPTION.
* With new extranal options it wil lbe easier to tweak evolution globally.

New EXTERNAL OPTIONS:

```
  {
    "type": "EXTERNAL_OPTION",
    "name": "MONSTER_UPGRADE_MIN_DAYS",
    "info": "Minimal days for monster upgrade.  Monster upgade time in days can't be lower than that no matter what.  This value doesn't scales with MONSTER_UPGRADE_FACTOR",
    "stype": "int",
    "value": 3
  }
```

Probably this should be available as mod for certain time. After that it can be obsolated and merged to base game. Even after that some polish likely will be requred. For example to set new upgrade times (mod mostly overrides it with external options).

**Comment by Firestorm01X2** [comment]
All zombies evolution time now bestween [10*4 ; 20*4]  (4 is default scaling evolution factor).

Also I've ddecided to keep zombie evolution factor on 4 to minimize changes.

**Comment by Firestorm01X2** [comment]
Likely it is conflicts with this:
https://github.com/CleverRaven/Cataclysm-DDA/pull/28090

So something has to be decided.

File: data/json/game_balance.json [hunk 1]
```diff
@@ -145,5 +145,12 @@
     "info": "Enables CBM slots mechanics. If true CBM slots are enabled.",
     "stype": "bool",
     "value": false
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "MONSTER_UPGRADE_MIN_DAYS",
+    "info": "Minimal days for monster upgrade.  Monster upgade time in days can't be lower than that no matter what.  This value doesn't scales with MONSTER_UPGRADE_FACTOR",
```
Comment (Night-Pryanik) [inline data/json/game_balance.json]:
```suggestion
    "info": "Minimal days for monster upgrade.  Monster upgrade time in days can't be lower than that no matter what.  This value doesn't scale with MONSTER_UPGRADE_FACTOR",
```

**Comment by Firestorm01X2** [comment]
Since solution was tested as mod - Ready.

**Comment by Firestorm01X2** [comment]
And yes one note about:
https://github.com/CleverRaven/Cataclysm-DDA/pull/27950#issuecomment-458317406

>The plan is for population numbers to be higher but the game starts with more mundane zombies at the start so that individuals are easier to deal with. Then over time evolution kicks in and ratchets up the difficulty.

I did notice part
>population numbers to be higher

PR DOES increase default zombie spawn a little. Zombe crowds in GROUP_ZOMBIE can twice as big. But:

1) Too hight zombe spawn may transform game in in slug fest without real increasing in difficulty.
2) Wander spawn already spawn addtional zombies.

That is why spawn number was increased not that high.

**Comment by kevingranade** [comment]
I don't see why you added a new option only to set it to some specific value, if we aren't actively tuning it it shouldn't be an option.

**Comment by Firestorm01X2** [comment]
>I don't see why you added a new option only to set it to some specific value, if we aren't actively tuning it it shouldn't be an option.

Done.

**Comment by Inglonias** [comment]
I like this one.

**Comment by SirPendrak** [comment]
Most things sound reasonable, but dont make the Zed ammout skyrocket too high, overall they should be arround same as beds number, otherwise it breaks immersion. And base, sugested evolution tempo should be increased, but thats probably up for debate. Many players can end with basic survivor gear, dodge skill of 7 and melee skill of ~4 after two weeks. 
This game could benefit much from making start much easier, and making mid and lategame much harder, and for that we need many, many more advanced evolutions, and probably higher ev. tempo.

**Comment by Firestorm01X2** [comment]
>.And base, sugested evolution tempo should be increased, but thats probably up for debate.

It is done here already, because of this:
>Enabled back true half life evolution time, that was changed here: #14077 . So now time to evolve will be more random, without unnecessary 100% pass interval.

This is basically means that evolution will be somewhat 2 times faster and a lot less predictible. It should compensate easier early game with basic zombes.

**Comment by Firestorm01X2** [comment]
>Why are you setting all half-life values to 20? Some types of zombie are meant to appear more slowly, the half-life is a balancing stat and shouldn't be the same for all zombie types.

1) Compensating easier game start.
2) Normalizing evolution time. Beacause 28*4 =112 days. It is more than 3 months. Overall half life time scatter was way too big between: 8*4 =32 and 28*4=112. Now it is between 10*4= 40 and 20 * 4 = 80. Other changes can be done via global evolution multiplier in game options.

However it it can be easily reverted and don't interfere with my idea much.

**Comment by I-am-Erk** [comment]
It's reasonable to see about rebalancing half-life times to be shorter, I just don't think they should be uniform. For example, when I made the kevlar2 soldier evolution, I set the half life to roughly double the previous entry because it more or less represents double the evolutionary changes. It's meant to be a pretty late arrival on the scene. Essentially, it skips an interval evolution and shows up with the next tier. I'm not saying evolution times are presently balanced, but they are intended as a balancing element, making the toughest zombies not show up at the start. (of course, this was mangled by people then adding their end-game zombies to starting monster groups for some reason).

Besides that, if you set all half-lives to universally 20, you may as well not have a half-life attribute.

**Comment by Firestorm01X2** [comment]
>Besides that, if you set all half-lives to universally 20, you may as well not have a half-life attribute.

Make sense. Reverting.

**Comment by Firestorm01X2** [comment]
Reverted. But I've set base zombe evolution time to 14 * 4 =56 days from 28 * 4 = 112 .
I need this to keep deiversity.

**Comment by Mcrone617** [comment]
A couple of questions, will this be implemented as a mod or just a base game change? Also are the newly added brainless zombies included in the day 1 spawns at all? It makes sense that there should be some walking around, especially with the extreme amount of firearms in the population. I might have missed this, so sorry if it's already in, my code reading skills aren't that good.

Can't wait for this to be merged, the era of running away from skeletal juggernauts on day one is almost over.

**Comment by Firestorm01X2** [comment]
Forgot new headless zombie. Need to add it.

**Comment by Firestorm01X2** [comment]
Removed some evolved zombies from other mapgens. It seeems that I ignored it.
Added evolution zombie dog-> skeleton dog.

**Comment by Firestorm01X2** [comment]
Ready.

**Comment by Firestorm01X2** [comment]
>A couple of questions, will this be implemented as a mod or just a base game change?

It is base game change.

**Comment by Firestorm01X2** [comment]
Looks like reloading gun test failing on Travis.
PR is ready.

**Comment by SirPendrak** [comment]
I have one personal wish. If gas bloated zombies are gonna be removed from butchers shop day 1 spawn, i am gonna be sad. Pls dont do this.

**Comment by kevingranade** [comment]
@SirPendrak need a reason.

**Comment by Firestorm01X2** [comment]
>I have one personal wish. If gas bloated zombies are gonna be removed from butchers shop day 1 spawn, i am gonna be sad. Pls dont do this.

It can be replaced with fat zombes that can evolve (and likely will) to bloated. It is same thing

**Comment by Firestorm01X2** [comment]
Ready. 
Thanks to @SirPendrak I've found some leftovers.

**Comment by Firestorm01X2** [comment]
Jeknkins, rebuild

**Comment by Firestorm01X2** [comment]
Jenkins, rebuild

**Comment by Mcrone617** [comment]
I know you probably can't do anything until its merged, but will #28944 mess with this at all? or is that a problem for when the skeletal brute is merged, and this PR doesn't have to worry about it.

**Comment by Firestorm01X2** [comment]
>I know you probably can't do anything until its merged, but will #28944 mess with this at all? or is that a problem for when the skeletal brute is merged, and this PR doesn't have to worry about it.

It is almost gurantee conflicts with current PR.

Looks like my PR pretty easy can catch conflicts.  It has almost 700 lines edited. 

It is better to merge it first. And preferably in nearest future.

**Comment by Firestorm01X2** [comment]
@Mcrone617 

I've looke your PR more precisely. The only thing that confilcs is likely monstergroups.json. To avoid that - just remove your brute from spawns. 

You monster is already step in the evolution of zombe skeleton. It should be enough. Skeletons will volves to brute naturally.

**Comment by Firestorm01X2** [comment]
I've forgot that removing monsters will afffect frequency so I've tird to merge firequency to monstergoups withoud default monster.

Additonally some spawns were corrected a little to mantain frequncy, closer to original spawns.

Ready.

**Comment by Inglonias** [comment]
I am kind of excited for this PR. Is there a specific reason it hasn't been merged yet, or does someone just need to get around to it?

**Comment by Night-Pryanik** [comment]
It's just huge amount of other PRs waiting for review and merge.

**Comment by Firestorm01X2** [comment]
And it catches confilct again.

**Comment by I-am-Erk** [comment]
It's also still labelled "WIP"

**Comment by Firestorm01X2** [comment]
>It's also still labelled "WIP"

I've set it myselft now.

**Comment by Firestorm01X2** [comment]
Ok. It is online again.
Ready

**Comment by Firestorm01X2** [comment]
Now brainless zombie evolves twice as fast. Based on player reports it is virually useless in it's basic form. And it is by design.
So let's make it evolve faster in something useful.

PR is still ready.

**Comment by TheMurderUnicorn** [comment]
Not to be rude, but I was in the process of fixing up the Brainless and Listener zombies when you asked me to stop and wait for this PR to resolve.

**Comment by Firestorm01X2** [comment]
>Not to be rude, but I was in the process of fixing up the Brainless and Listener zombies when you asked me to stop and wait for this PR to resolve.

No problem. I can revert my change. It is just a digit now

**Comment by Firestorm01X2** [comment]
>Not to be rude, but I was in the process of fixing up the Brainless and Listener zombies when you asked me to stop and wait for this PR to resolve.

My change was reverted.

**Comment by TheMurderUnicorn** [comment]
It wasn't that big a deal.  I was just saying that the change was literally the same as part of my PR.  I've been checking this thread daily, waiting for this PR to resolve so that I can 'fix' the brainless and listeners, that's all.

**Comment by I-am-Erk** [comment]
> It wasn't that big a deal. I was just saying that the change was literally the same as part of my PR. I've been checking this thread daily, waiting for this PR to resolve so that I can 'fix' the brainless and listeners, that's all.

I'd recommend balancing your own currently active submissions. This is a very large PR and is going to need significant testing before merging.

**Comment by Mcrone617** [comment]
> > It wasn't that big a deal. I was just saying that the change was literally the same as part of my PR. I've been checking this thread daily, waiting for this PR to resolve so that I can 'fix' the brainless and listeners, that's all.
> 
> I'd recommend balancing your own currently active submissions. This is a very intensive PR and is going to need significant testing before merging.

Is there a way to help along the testing process? I've been holding off for a couple of weeks now on a new campaign in anticipation of this PR, and I would definitely be willing to download a test version of the game with these PR changes to help out ( I know very little about Github, would something like this be possible?) 
By testing do you mean just playing a couple of basic games and seeing how hard/easy it is to get to late game? or how quickly evolution kicks in, stuff like that.

**Comment by I-am-Erk** [comment]
Sure, anyone can leave a review approving or suggesting changes. You'd just have to download the edited files and test them out, or review based on the text content of the changes if you know how to read json (although it's balance effects I'm most concerned about which are best managed through actual play).

**Comment by Firestorm01X2** [comment]
> read json (although it's balance effects I'm most concerned about which are best managed through actual play).

There is cpp code change. Minimal, but important.

**Comment by Firestorm01X2** [comment]
@Mcrone617 
@I-am-Erk 

https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/06.04.2019
Just download and play.
Here is built version this PR. 

Just ignore 0.C version mark - it is 0.D Experimental from yesterday.

**Comment by Mcrone617** [comment]
> @Mcrone617
> 
> 
> https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/06.04.2019
> Just download and play.
> Here is built version this PR.
> 
> Just ignore 0.C version mark - it is 0.D Experimental from yesterday.
> 
> Saves should be compatible with usual and future experimentals.

Thanks, I'm pretty busy IRL right now, but i'll try to sneak a couple of games in over the next week and report back my findings. I'm stopping a character when I reach the "endgame" and what I define as the endgame in this case is having a working car, solar/renewable base of some sort, a forge set up, and owning the required books needed to get to max skills in stuff like fabrication or mechanics. I'm also planning on playing as a default survivor, and at least playing until summer (3 months/90 days) each character, to see the new evolution in effect. Is there anything else needing testing?

Also, @Firestorm01X2  , would you be alright with me posting this thread on the CDDA reddit? It would probably get at least some people to test it, especially since you made it easy with a download link. The more the merrier as far as balancing goes. But no worries if you don't want too/you want to post it yourself, just let me know.

**Comment by Firestorm01X2** [comment]
@Mcrone617 
There is option to "rewind" current time in debug menu. It will simplify testing a lot.

**Comment by Mcrone617** [comment]
> 
> > Also, @Firestorm01X2 , would you be alright with me posting this thread on the CDDA reddit?
> 
> Yes. I am ok with that. Link is public and intended for everyone want to try it.

Here's the link:
https://www.reddit.com/r/cataclysmdda/comments/ba6xic/zombie_evolution_rework_needs_testers/

hopefully some people help out with the testing.

**Comment by Firestorm01X2** [comment]
Updated version uploaded:
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/07.04.2019

It was just synced with master.

**Comment by Firestorm01X2** [comment]
Updated version uploaded:
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/10_04_2019

Sync with master

**Comment by Mcrone617** [comment]
Alright, I just finished my first game, playing a completely new character up to the 30 day mark, and I've got some feedback and photos to help spur analysis. I played as a default survivor, evac shelter, although I did orient him towards combat, with traits like pain resistance and whatnot, but nothing crazy stats-wise, 10 across the board. I spawned near a small/medium sized town, with a decent amount of zombies about. Code-wise it seems to be working as intended, as the large majority of zombies are un-mutated zombies.
Firestorm PR testing G1 P4 Character stats
Firestorm PR testing G1 P2 overmap
I spent about the first 15 days completely clearing out Arlington, which admittedly was much easier due to the large majority of zombies being basic, Being able to kite zombies into windows without having to worry about hulks destroying everything made it easier on town cleanup. The evolution is in place however, as on night one I had a smoker zombie chasing me as I went for a shopping cart in town, I don't remember seeing it before, so it probably evolved almost immediately after the game started. Once I finished off the town, I spent a couple of days looting it, and found books/grinded enough to get my fabrication up to forging level. I then created decent mid-game gear, such as a steel spear, and made a expedition to Newmarket, up north. I ended the game after spending a couple of days in Newmarket thinning hordes, and checking their composition. Here is a example of what I found on day 28.
Firestorm PR testing G1 P1 horde
As you can see there is still a decent amount of normal zombies, although interestingly enough an corrosive zombie and a screecher zombie. This is interesting because I hadn't seen a spitter zombie or a shrieker zombie at all yet. It's possible that a spitter zombie evolved and then it got selected to evolve again, but that seems quite unlikely, so it may be worth checking out the weights that the later zombie are selected to evolve. It might be better for most of the basic zombie to evolve once before the endgame stuff (2nd stage evolution) is prioritized.
Ultimately, here was my kill list, and you can see there was a decent scattering of evolved zombies that appeared in it.
Firestorm PR testing G1 P3 Kills
For overall feedback, I would say that this PR is good, but probably needs a couple of more tweaks. I did very well and managed to clear a town in a couple of weeks, but I know how to play the game decently well, and I rushed effective weaponry, newer players might have trouble knowing when to retreat and using choke-points. Still I would argue that zombie spawn needs to be turned up very slightly again, as it was pretty easy to kill hordes when there aren't specials. I would also recommend looking into selection weights for evolution for 2nd tier zombies, should they be appearing on day 30 surrounded by a horde of normal zombies? 
I am planning to do more data-driven testing now, as this was a good test, it was still only one test. I plan to alter the game time via debug and check various towns at the 1 month/2 month mark and see what zombie evolution has occurred on multiple games, and I will report back on my findings, as the above game could have just been out of the ordinary. I then plan to play another characters once I report my findings on the debug testing, please let me know if there are any specific things I should focus on, maybe do a game to the 2 month mark?
TLDR: the PR is pretty good at creating the feel outlined here, although I personally feel it is a bit easy, more testing is required.
"I am suriving for one month and I am seeing that zombies are changing. I am starting to see very werid creatures in the horde from time to time."
Start is easier- mostly only mundane zombies on the streets. But it is not going to be that way forever - you will see new mosnters at the end of the firts month. Not a much of them, however.
First mounth you have to deal with only mundane zombies mostly. Skeletal Juggernout actually can happen in first month but it should be very unlikely. But still - beware.

**Comment by Firestorm01X2** [comment]
@Mcrone617
Really thank you for review. Ii sreally helpful.

>As you can see there is still a decent amount of normal zombies, although interestingly enough an corrosive zombie and a screecher zombie. This is interesting because I hadn't seen a spitter zombie or a shrieker zombie at all yet. It's possible that a spitter zombie evolved and then it got selected to evolve again, but that seems quite unlikely,

In fact normal zombie can evolve to corrisive directly.
 "name": "GROUP_ZOMBIE_UPGRADE",
{ "monster": "mon_zombie_corrosive", "freq": 10, "cost_multiplier": 7 },

It is in the evolution logic.
Yes sometimes evoltuin paths still intersects. It is exacly the case.

>Still I would argue that zombie spawn needs to be turned up very slightly again, as it was pretty easy to kill hordes when there aren't specials.

I've expected for easier game in the beginning. It is part of the plan.

About quantity vs quality. Defintely just adding a lot more zombies will make game not harder but more sluggish. I want to avoid that. 

Actually I've expected number of zombies to be higher after change.  I even have report about that on other resource. Reason- removing high level zombies with hight price multiplier from spawn leave more place for usual ones. So there is should be already somewhat more zombies.

Maybe default global evolution rate can be changed to achive more difficulty. Or yes - time for evolution for weaker evolved zombies can be decreased. But it is very debatable and it is not straightforward to  determine what exaclty to change.

**Comment by Firestorm01X2** [comment]
@Mcrone617 
> I then plan to play another characters once I report my findings on the debug testing, please let me know if there are any specific things I should focus on, maybe do a game to the 2 month mark?

Probably is is better to just continue playing and give reports if game become harder on second month.
Debug testing is definetely can be used.

**Comment by Firestorm01X2** [comment]
Merged conflicts with:
https://github.com/CleverRaven/Cataclysm-DDA/pull/28944

Good thin that new Skeleton brute is part of skeleton evolution. So he does not need to be spawned at start or added in upgade group.

**Comment by Firestorm01X2** [comment]
Incoming coflicts from:

Thorny Shamblers:
https://github.com/CleverRaven/Cataclysm-DDA/pull/29406

Gasoline zombies:
https://github.com/CleverRaven/Cataclysm-DDA/pull/29307

**Comment by Firestorm01X2** [comment]
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/14_04_2019_2
* Updated for 14.04.2019. Say "hello!" to new skeleton brutes and explosive zombies.
* Reduced chance of evolving default zombies to just tought zombes. It leads to more varieaty in evolution quicker.
* No more groups of 7 shocker zombies. Fixed spawn chances.
* Increased chance of evolving fat zombies to boomers. Was too many bloated zombies.
* Increased vairety in childen evolution.
* Removed direct evolution to corrosive zombies and huge boomer.

**Comment by Firestorm01X2** [comment]
Evolution rework for testing purposes.
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/20_04_2019

* Updated for 20.04.2019. New Thorny Shambler included. It does not appear in beginning of the game but will come as evolution of normal zombies.
* Removed direct evolution to zombie predator. Hunter will evolve to predator as before.

**Comment by Firestorm01X2** [comment]
Now we have this:

https://www.somsubhra.com/github-release-stats/?username=Firestorm01X2&repository=Cataclysm-DDA

> Total Downloads 25

No reports about problem here:
https://www.reddit.com/r/cataclysmdda/comments/ba6xic/zombie_evolution_rework_needs_testers/

So it is ready. It is definitely time to merge this.

**Comment by Mcrone617** [comment]
I did some testing on the build that @Firestorm01X2 just released with the thorny shamblers, and i'm inclined to agree, except for one thing I think I found though debug testing. I took some screenshots for a summary of what I found each testing timeframe.
Day 30/1 month:
Day_30
This is basically what I found on the first month in my game, so not much has changed here (which is good) a decent scattering of the t1 specials, and a t2 special here and there.
Day 60/2 months:
Day_60
Mostly the same as the 1st month, although more specials are appearing. You can see that the T1 specials are beginning to meet the level of the normal zombies, similar to how zombie evolution is at the start of the game right now.
Season Summer/3 months:
Day_Summer
As you can see the specials are beginning to outnumber the normal zombies at this point, similar to 2 month, and how it currently is in evolution.
Season Fall/6 months:
Day_Fall
I jumped ahead to the start of fall, and things are starting to get intense as far as evolution goes, the t3 Hulk enemies are sppearing consistently, and  the super-dangerous hordes are beginning to develop (as seen in the bottom left) Of note, this was the first time I saw a thorny shambler, so I'd say that's pretty balanced.
Season Winter/9 months:
Day_W
I started to see a couple of necromancers with hordes, and in general the hordes continued to get stronger, Lot more T2 specials now.
Year 2:
Day_Y2
Day_Y2_2
Things are starting to get crazy, basically no normal zombies, kevlar hulks, mutiple necormancers, corrosive zombies, etc. Gasoline zombies started to appear around here, and they're still pretty rare too. which is good for concerns about them completely nuking towns with fire.
Year 3:
Day_Y3_1
Like before, just more specials keep appearing. Masters start to appear consistently with hordes from my limited testing, before they were slightly rare, but because they appear later, they can be the "clean=up" function of the horde, any zombie who was skipped by zombie evolution can be quickly upgraded by the master, as opposed to their current role of quickly making hordes impossible to beat early on.
Year 4:
Day_Y4_1
Basically everything is upgraded at this point, you better be a superman mutant cyborg to take this on.
Year 10:
Day_Y10_1
I did a year 10 for the fun of it, it's pretty maxed out (unsurprisingly) and doesn't seem to change that much from year 3 or 4.  

However, if you noticed from my photos, in my hour or so of testing I did not run into one specific zombie that is pretty important in my view. The default Hulk did not appear once during my testing. All the special types of Hulks did (skeletal, Kevlar, nightstalker, etc) the zombie wrestler appeared, and the zombie brute appeared, but for some reason, I never saw a default hulk at all. Possibly there is something preventing them from evolving from brutes? I would check that out firestorm, but once you fix that/clarify that I would agree with you in saying that it's ready for merging. aside from Hulks not appearing, it seemed very balanced, and provides a nice "scale" for the player to climb with, and with the eventual addition of stuff like the stamina re-balance, making clearing hordes harder, this should help compensate for that.

**Comment by Firestorm01X2** [comment]
> I would check that out firestorm, but once you fix that/clarify that I would agree with you in saying that it's ready for merging. aside from Hulks not appearing, it seemed very balanced, and provides a nice "scale" for the player to climb with, and with the eventual addition of stuff like the stamina re-balance, making clearing hordes harder, this should help compensate for that.

Hulk evolution chance are quite low. 0.7%. It is definitely must be changed.  On it.

**Comment by Firestorm01X2** [comment]
Fixed. The only thing that was changed - is chance to evlove to hulk.
Ready.

**Comment by Firestorm01X2** [comment]
https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/20_04_2019_2

* Updated for 20.04.2019 again. 
* Fixed  extremly low chance of eloving Brute to Hulk (0.7% -> 10%).

**Comment by Mcrone617** [comment]
> https://github.com/Firestorm01X2/Cataclysm-DDA/releases/tag/20_04_2019_2
> 
> * Updated for 20.04.2019 again.
> * Fixed  extremly low chance of eloving Brute to Hulk (0.7% -> 10%). 

https://www.reddit.com/r/cataclysmdda/comments/ba6xic/zombie_evolution_rework_needs_testers/

I've updated the Reddit thread with the new build, but because of the fix now, I'd say it's ready to be merged. If any of the devs/mergers have some additional concerns still that I didn't address above, please let me know and I'd be happy to test it. Additionally, Firestorm thanks for letting me know about that download tool, I had no idea and it will be pretty helpful to see how many times a PR has been tested. Lets get this rework merged!

**Comment by Inglonias** [comment]
Congratulations. This is a big step to balacing the game's difficulty curve!

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/request-to-developers/22170/1
